### PR TITLE
feat(backend): log streaming

### PR DESCRIPTION
**Pull request by HurricanKai**
Continuation of https://github.com/windmill-labs/windmill/pull/776

Reworks log streaming to be based on channels & streams. This leads to stream handling being more centralized & performance improvements, especially around large logs.
There are further improvements to do here, but these give a good basis to do further testing.

(Bench in Progress)

### CODE REVIEW

File: backend/src/worker.rs [hunk 1]
```diff
@@ -492,7 +494,45 @@ async fn handle_queued_job(
             .await?;
         }
         _ => {
-            let mut logs = "".to_string();
+            let (tx, mut rx) = mpsc::unbounded_channel::<String>();
```
Comment (rubenfiszel) [inline backend/src/worker.rs:497]: let's rename for clarity: logs_tx, logs_rx

File: backend/src/worker.rs [hunk 2]
```diff
@@ -1491,7 +1551,7 @@ async fn handle_dependency_job(
         Err(error) => {
             sqlx::query!(
                 "UPDATE script SET lock_error_logs = $1 WHERE hash = $2 AND workspace_id = $3",
-                &format!("{logs}\n{error}"),
```
Comment (rubenfiszel) [inline backend/src/worker.rs:1494]: this is a feature change, we still want to display the logs here

File: backend/src/worker.rs [hunk 3]
```diff
@@ -1741,21 +1802,15 @@ async fn handle_child(
     /* a future that reads output from the child and appends to the database */
     let lines = async move {
         /* log_remaining is zero when output limit was reached */
-        let mut log_remaining = (MAX_LOG_SIZE as usize).saturating_sub(logs.chars().count());
+        let mut log_remaining = MAX_LOG_SIZE as usize - 1000;
```
Comment (rubenfiszel) [inline backend/src/worker.rs:1805]: - 1000 ?

File: backend/src/worker.rs [hunk 4]
```diff
@@ -492,7 +494,45 @@ async fn handle_queued_job(
             .await?;
         }
         _ => {
-            let mut logs = "".to_string();
+            let (tx, mut rx) = mpsc::unbounded_channel::<String>();
+            let (btx, brx) = broadcast::channel(512);
+            tokio::spawn(async move {
+                let btx = btx;
+                while let Some(e) = rx.recv().await {
+                    for e in e.split('\n') {
```
Comment (HurricanKai) [inline backend/src/worker.rs:502]: remove this - inline the channel

**Comment by HurricanKai** [comment]
Putting this as a draft for now, as we don't wanna merge it as is - some extra work to move log aggregation to postgres & removing some unecessary streams is needed.
I don't want to make those changes until I've added log tests though, which need to happen in another branch.

**Comment by HurricanKai** [comment]
Closing this as the branch is very out of data. I'm keeping the branch around and still want to land this work once I have time to get back to perf work.

**Pull request closed without merging**